mongofiles/mongofiles.go

```diff
@@ -467,8 +495,33 @@ func (mf *MongoFiles) Run(displayHost bool) (output string, finalErr error) {
 	case Get, GetID:
 		err = mf.handleGet()
 
-	case Put, PutID:
-		err = mf.handlePut()
+		// If mongofiles --put ... is called, i.e. with multiple supporting
+		// arguments, then add gridFiles specified in mf.FileNameList
+		// (see TOOLS-2667). Otherwise, if mongofiles --put_id is called, then
+		// preserve existing behavior.
+	case Put:
+		for _, file := range mf.FileNameList {
+			id, err := mf.parseOrCreateID()
+			if err != nil {
+				return "", err
+			}
+
+			err = mf.handlePut(id, file)
+			if err != nil {
+				return "", err
+			}
```

Comment:
The body of this for loop is identical to the case below (`case PutID`) except for the filename argument to `mf.handlePut`. It seems unnecessary to argue `id` to `handlePut`. You can instead have the `id, err := mf.parseOrCreateID()` logic inside the `handlePut` method and have `handlePut` only take the filename as an argument. Specifically:
```
func (mf *MongoFiles) handlePut(filename string) error {
    id, err := mf.parseOrCreateID()
    if err != nil { ... }
    
    ...
}

...

    case Put: 
        for _, filename := range mf.FileNameList {
            err = mf.handlePut(filename)
            if err != nil { ... }
        }
    case PutID:
            err = mf.handlePut(filename)
            if err != nil { ... }
```

Reply:
Makes sense -- consider it done!

Reply:
Consider it done.